src/modules/hip/kernel/slice.hpp

```diff
@@ -32,79 +87,487 @@ __global__ void slice_ncdhw_hip_tensor(T *srcPtr,
+                               dstPtrTemp,
+                               make_uint3(0, 0, dstGenericDescPtr->strides[1]),
+                               1,
+                               make_uint3(1, maxHeight, maxWidth));
         }
     }
-    else if (dstGenericDescPtr->layout == RpptLayout::NDHWC)
+    else if(numDims == 1)
     {
-        int globalThreads_x = (dstGenericDescPtr->strides[2] / 3 + 7) >> 3; // W - width (x direction) - vectorized for 8 element loads/stores per channel
-        int globalThreads_y = dstGenericDescPtr->dims[2];               // H - height (y direction)
-        int globalThreads_z = dstGenericDescPtr->dims[1];               // D - depth (z direction)
+        int globalThreads_x = (dstGenericDescPtr->strides[0] + 7) >> 3; // W - width (x direction) - vectorized for 8 element loads/stores per channel
+        int globalThreads_y = 1;
+        int globalThreads_z = 1;
 
         for(int batchCount = 0; batchCount < dstGenericDescPtr->dims[0]; batchCount++)
         {
-            hipLaunchKernelGGL(slice_ndhwc_hip_tensor,
+            Rpp32s *anchor = &anchorTensor[batchCount * numDims];
+            Rpp32s *shape = &shapeTensor[batchCount * numDims];
+            Rpp32u *roi = roiTensor + batchCount * numDims * 2;
+            Rpp32s *length = reinterpret_cast<Rpp32s *>(&roi[numDims]);
+            Rpp32u maxLength = std::min(shape[0], length[0] - anchor[0]);
+            T *srcPtrTemp = srcPtr + (batchCount * srcGenericDescPtr->strides[0]) + anchor[0];
+            T *dstPtrTemp = dstPtr + (batchCount * dstGenericDescPtr->strides[0]);
+
+            hipLaunchKernelGGL(slice_ncdhw_hip_tensor,
                                dim3(ceil((float)globalThreads_x/LOCAL_THREADS_X), ceil((float)globalThreads_y/LOCAL_THREADS_Y), ceil((float)globalThreads_z/LOCAL_THREADS_Z)),
-                               dim3(LOCAL_THREADS_X, LOCAL_THREADS_Y, LOCAL_THREADS_Z),
+                               dim3(LOCAL_THREADS_X, 1, 1),
                                0,
                                handle.GetStream(),
-                               srcPtr + (batchCount * srcGenericDescPtr->strides[0]),
-                               make_uint2(srcGenericDescPtr->strides[1], srcGenericDescPtr->strides[2]),
-                               dstPtr + (batchCount * dstGenericDescPtr->strides[0]),
-                               make_uint2(dstGenericDescPtr->strides[1], dstGenericDescPtr->strides[2]),
-                               &roiGenericPtrSrc[batchCount]);
+                               srcPtrTemp,
+                               make_uint3(0, 0, 1),
+                               dstPtrTemp,
+                               make_uint3(0, 0, 1),
+                               1,
+                               make_uint3(1, 1, maxLength));
+        }
+    }
+
+    return RPP_SUCCESS;
+}
+
+template <typename T>
+RppStatus hip_exec_fill_value_tensor(T *dstPtr,
```

Comment:
Suggest to always place all dependencies above the function that calls those deps. Since this is a dependency for hip_exec_slice_tensor(), place it right above it.

Reply:
Done
Again, it is somehow not showing here